MergerService/Utils/TaskUtils.cs

```diff
@@ -125,9 +126,9 @@ public void UpdateReject(string jobId, string taskId, int attempts, string reaso
                 attempts++;
 
                 // Check if the task should actually fail
-                if (!resettable || attempts == this._maxAttempts)
+                if (managerCallbackUrl is not null && (!resettable || attempts == this._maxAttempts))
```

Comment:
managerCallbackUrl can be null, why did you add it here?

Reply:
because it shouldn't update failed, it uses the callback URL, but it should update the job - for exporter situations.
i consult with Shimon